### PR TITLE
23.x rebase: crypto & validation

### PR DESCRIPTION
**Pull request by vertiond**

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -483,8 +483,34 @@ crypto_libbitcoin_crypto_base_a_SOURCES = \
   crypto/sha3.h \
   crypto/sha512.cpp \
   crypto/sha512.h \
+  crypto/scrypt.cpp \
+  crypto/scrypt.h \
```
Comment (cruelnovo) [inline src/Makefile.am:487]: How come crypto/scrypt_sse2.cpp was removed?  I see the scrypt_N_1_1_256_sp_sse2() function is still called in scrypt.cpp...

Comment (vertiond) [reply src/Makefile.am:487]: I see.. yeah there wasn't any good reason for not including it.

Comment (vertiond) [reply src/Makefile.am:487]: added it back

**Review by cruelnovo — APPROVED** [review]
LGTM! I have reviewed this PR by comparing it to an independent rebase and the changes seem to be in order! I approve the merge of this PR.

**Pull request merged**